format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSink.java

```diff
@@ -76,8 +96,17 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   public void prepareRun(BatchSinkContext context) throws Exception {
     FailureCollector collector = context.getFailureCollector();
     config.validate(collector);
-    ValidatingOutputFormat validatingOutputFormat = context.newPluginInstance(FORMAT_PLUGIN_ID);
     FileFormat fileFormat = config.getFormat();
+    ValidatingOutputFormat validatingOutputFormat;
+    try {
+      validatingOutputFormat = context.newPluginInstance(fileFormat.name().toLowerCase());
+    } catch (InvalidPluginConfigException e) {
+      String errorMessage = String.format("Format '%s' cannot be used because properties %s were not provided or " +
+                                          "were invalid %s when the pipeline was deployed. Set the format to a " +
```

Comment:
e.getInvalidProperties() is a set of objects, not property names. It will not display nicely. The location of that third '%s' is also awkward, it won't be a good sentence. 

Let's remove that last %s

invalid %s -> invalid

and create a union of the property names from missing properties and invalid properties.

Reply:
also, indentation here is inconsistent. Continuing a statement on the next line should be a 2 space indent, not aligned with the previous line. alignment is used when there are multiple arguments for a single method call spread out across multiple lines.

```
format("message '%s' " + 
         "wraps ",
       arg);
```

please see rest of the project for examples, and update your IDE accordingly.

Reply:
fixed indentation.